service/app/controllers/otp/validator/OtpRequestValidator.java

```diff
@@ -46,12 +48,13 @@ private void validateTypeAndKey(Request otpRequest) {
     } else if (JsonKey.PHONE.equalsIgnoreCase(type)) {
       validatePhone(key);
     } else {
-      throw new ProjectCommonException(
-          ResponseCode.invalidParameterValue.getErrorCode(),
-          ResponseCode.invalidParameterValue.getErrorMessage(),
-          ResponseCode.CLIENT_ERROR.getResponseCode(),
-          type,
-          JsonKey.TYPE);
+      ProjectCommonException.throwClientErrorException(
+          ResponseCode.invalidValue,
+          ProjectUtil.formatMessage(
+              ResponseCode.invalidValue.getErrorMessage(),
+              JsonKey.TYPE,
+              type,
+              String.join(StringFormatter.COMMA, JsonKey.EMAIL, JsonKey.PHONE)));

```

Comment:
if there is a relevant functional test. - we need to change that as well.
please raise a PR for that as well.